### ISSUE TITLE
Support for document versions in ingester pipelines

### ISSUE THREAD

**Issue by geneqew** [elastic/elasticsearch#27242]
**Describe the feature**:
Document versioning is not supported when indexing documents through ingester pipelines. One possible use case would be log entries are formatted in JSON where one of it's attributes indicate the document version. Since org.elasticsearch.ingest.PipelineExecutionService uses an IndexRequest, it might be helpful to pair this with a version processor to indicate which field of the data being captured can be used as the document version and which versioning type to use.

**Comment by martijnvg** [comment]
Adding the ability to access and change the version from ingest makes sense. This should be similar to how _index and _id meta fields are being used: https://www.elastic.co/guide/en/elasticsearch/reference/5.6/accessing-data-in-pipelines.html#accessing-metadata-fields

So there is no need for a new processor:

```
{
  "set": {
    "field": "_version"
    "value": "{{my_field_containing_version}}"
  }
}
```

**Comment by geneqew** [comment]
The reason i suggested the use of a separate processor is to simplify the setting of the _version field while at the same time giving the flexibility to choose the [version type](https://www.elastic.co/guide/en/elasticsearch/reference/current/docs-index_.html#_version_types). Something like:

```
{
  "version" : {
     "field" : "{{my_field_containing_version}}",
     "type" : "external_gte" // if not provided defaults to 'external'
  }
} 
```

Otherwise the suggestion to use the set processor should be good enough to support external versions given that there is no need for an 'external_gte' version type.

**Comment by martijnvg** [comment]
I think the `version_type` can be made available too as a meta field. So that then can be set via a set processor too:

```
{
  "set": {
    "field": "_version_type"
    "value": "external_gte"
  }
}
```

However then in the case you're describing you would need two processor, so I think it makes sense to have a dedicated processor for this. This then also allows ingest the for example validate the provided version type and fail on pipeline creation if it is invalid.

**Comment by martijnvg** [comment]
This has been implemented via #27573
